libtenzir/builtins/operators/python.cpp

```diff
@@ -207,7 +203,23 @@ class python_operator final : public crtp_operator<python_operator> {
         }
         return caf::detail::scope_guard([maybe_venv] {
           if (maybe_venv) {
-            std::filesystem::remove_all(*maybe_venv);
+            std::error_code ec;
+            auto exists = std::filesystem::exists(*maybe_venv, ec);
+            if (ec) {
+              // ctrl can already be gone, so we can't emit a diagnostic here.
+              TENZIR_WARN("python operator failed to remove venv at {}: {}",
+                          *maybe_venv, ec);
+              return;
+            }
+            if (!exists) {
+              return;
+            }
+            std::filesystem::remove_all(*maybe_venv, ec);
```

Comment:
You can add a log message here to determine whether we call `remove_all` more than one time.

Reply:
We just tried this locally and the scope guard is only called once.